backend/src/main/java/eu/bbmri_eric/negotiator/notification/UserNotificationServiceImpl.java

```diff
@@ -169,6 +189,70 @@ public void notifyUsersAboutNewPost(Post post) {
     }
   }
 
+  @Override
+  public String getNotificationTemplate(String templateName) {
+    log.info("Getting notification template.");
+    validateTemplateName(templateName);
+    try {
+      org.springframework.core.io.Resource resource =
+          resourceLoader.getResource(thymeleafPrefix + templateName + thymeleafSuffix);
+      return new String(resource.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      log.error("Failed to get notification template", e);
+      throw new EntityNotFoundException(templateName);
+    }
+  }
+
+  @Override
+  public String updateNotificationTemplate(String templateName, String template) {
+    log.info("Updating notification template.");
+    if (thymeleafPrefix.equals(defaultThymeleafPrefix)) {
+      throw new ForbiddenRequestException("Cannot update default templates");
+    }
+    validateTemplateName(templateName);
+    String validateTemplate = validateHtml(template);
+    String targetTemplatePath = thymeleafPrefix + templateName + thymeleafSuffix;
+
+    org.springframework.core.io.Resource defaultTemplateResource =
+        resourceLoader.getResource(targetTemplatePath);
+
+    if (!defaultTemplateResource.exists()) {
+      throw new ForbiddenRequestException(
+          "Default template for " + targetTemplatePath + " does not exist");
+    }
+
+    writeTemplateToFile(targetTemplatePath, validateTemplate);
+    return validateTemplate;
+  }
+
+  @Override
+  public String resetNotificationTemplate(String templateName) {
+    log.info("Resetting notification template.");
+    if (thymeleafPrefix.equals(defaultThymeleafPrefix)) {
+      throw new ForbiddenRequestException("Cannot update default templates");
+    }
+    validateTemplateName(templateName);
+    String defaultTemplatePath = "classpath:/templates/" + templateName + thymeleafSuffix;
+    org.springframework.core.io.Resource defaultTemplateResource =
+        resourceLoader.getResource(defaultTemplatePath);
+
+    if (!defaultTemplateResource.exists()) {
+      throw new EntityNotFoundException(defaultTemplatePath);
+    }
+    try (InputStream inputStream = defaultTemplateResource.getInputStream()) {
+      String defaultTemplate = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+
+      String targetTemplatePath = thymeleafPrefix + templateName + thymeleafSuffix;
+      writeTemplateToFile(targetTemplatePath, defaultTemplate);
+
+      return defaultTemplate;
+
+    } catch (IOException e) {
+      log.error("Failed to reset notification template", e);
+      throw new RuntimeException("Failed to reset notification template", e);
```

Comment:
imho this is mixing the business logic with retrieval logic. not sure the Service should care where the templates are stored

Reply:
this was separated